OpenRA.Game/Traits/Player/Shroud.cs

```diff
@@ -172,18 +183,14 @@ void Invalidate(IEnumerable<PPos> changed)
 					type = count > 0 ? ShroudCellType.Visible : ShroudCellType.Fog;
 				}
 
+				oldResolvedType[uv] = resolvedType[uv];
 				resolvedType[uv] = type;
+				if (type != oldResolvedType[uv])
+					OnShroudChanged(uv);
 			}
 
-			if (CellsChanged != null)
-				CellsChanged(changed);
-
 			var oldHash = Hash;
-			Hash = Sync.HashPlayer(self.Owner) + self.World.WorldTick * 3;
-
-			// Invalidate may be called multiple times in one world tick, which is decoupled from rendering.
-			if (oldHash == Hash)
-				Hash += 1;
+			Hash = Sync.HashPlayer(self.Owner) + self.World.WorldTick;
```

Comment:
Was there a specific reason to drop the `* 3` here?

Reply:
Because this is now called only once per tick by definition, the owner + tick are by themselves enough to have a unique hash. The `* 3` seems to have only been added to make room for the `+= 1` s.